spec/truffle/interop/keys_spec.rb

```diff
@@ -11,6 +11,22 @@
 
 describe "Truffle::Interop.keys" do
   
+  it "returns an array" do
+    keys = Truffle::Interop.keys({'a' => 1, 'b' => 2, 'c' => 3})
+    keys.should be_an_instance_of(Array)
+  end
+  
+  it "returns an array of Ruby strings by default" do
+    keys = Truffle::Interop.keys({'a' => 1, 'b' => 2, 'c' => 3})
+    keys[0].should be_an_instance_of(String)
+  end
+  
+  it "returns an array of Java strings if you don't use conversion" do
+    keys = Truffle::Interop.keys_without_conversion({'a' => 1, 'b' => 2, 'c' => 3})
+    key = keys[0]
```

Comment:
This doesn't convert to a Ruby String?
I thought you'd need the `read_without_conversion` below. What is it for actually?

Reply:
No this doesn't convert to a Ruby string. It's not a foreign read. `read_without_conversion` is leftover from debugging something else. I'll document it and leave it in in case it's useful somewhere else.

Reply:
Ah, because we return a Ruby Array, right.
But if we returned say a TruffleObject array, then we would have to avoid conversion on the read too.

Reply:
We can't return a Java array, as we don't always have Java interop available. We'd have to define our own new array object with a new message resolution etc etc. I don't see any problem at the moment.

Reply:
It's fine as-is.
I think SVM supports JavaInterop.asJavaObject (or well, the new API since this got deprecated) and only cuts the more complex cases, but not sure.